### ISSUE TITLE
How can I determine which polygon region is being triggered by mouse hover?

### ISSUE THREAD

**Issue by FerretAngel** [MelihAltintas/vue3-openlayers#327]
Polygon regions cannot be bound with an ID, and I have searched through the documentation but have not found a way to add IDs to polygon regions. This means that I cannot accurately determine which polygon is being selected through the map's pointermove event.
![image](https://github.com/MelihAltintas/vue3-openlayers/assets/58507654/3dc6f6ca-9f60-4201-8013-399409c4ef8d)
![image](https://github.com/MelihAltintas/vue3-openlayers/assets/58507654/561d6579-3dee-45ce-8bcc-b8f0a2a9c276)
![image](https://github.com/MelihAltintas/vue3-openlayers/assets/58507654/a28aec14-3577-4aad-a2f3-d4a1d4a40078)

**Comment by FerretAngel** [comment]
My requirement is to map the feature objects with my own data IDs. How can I bind them together?

**Comment by FerretAngel** [comment]
I have encapsulated a polygon component by adding an id option for polygons myself. The usage is basically the same.

Usage:
```vue
<Polygon
    :id="item.id"
    :coordinates="item.lnglats"
    :fill="item.style.fill"
    :stroke="item.style.border"
    />
```
```vue
<script lang="ts" setup>
import { Feature } from "ol";
import type { Vector } from "ol/source";
import type { Coordinate } from "ol/coordinate";
import { Polygon } from "ol/geom";
import { type PropType, inject, onMounted, onUnmounted, type Ref } from "vue";
import { Stroke, Fill, Style } from "ol/style";
const { value: source } = inject("vectorSource") as Ref<Vector>;
const {
  coordinates,
  id,
  stroke: _stroke,
  fill: _fill,
} = defineProps({
  coordinates: {
    type: Array as PropType<Coordinate[]>,
    required: true,
  },
  id: {
    type: Number,
    default: Math.random(),
  },
  stroke: {
    type: Object as unknown as PropType<
      ConstructorParameters<typeof Stroke>[0]
    >,
  },
  fill: {
    type: Object as unknown as PropType<ConstructorParameters<typeof Fill>[0]>,
  },
});
if (!source) throw new Error("source is required");
const polygon = new Polygon([coordinates]);
const feature = new Feature({
  geometry: polygon,
});
feature.setId(id);
const stroke = new Stroke(_stroke);
const fill = new Fill(_fill);
feature.setId(id);
feature.setStyle(new Style({ stroke, fill }));
onMounted(() => {
  source.addFeature(feature);
});
onUnmounted(() => {
  source.removeFeature(feature);
});
</script>

<template></template>

```

**Comment by d-koppenhagen** [comment]
Please provide a minimal reproduction example.
Ideally, you can fork the [Stackblitz starter template](https://stackblitz.com/edit/vue3-openlayers-template) for this purpose.
From your code, I cannot extract what the `polygons` are and if they have an `id`.
If you want to somehow iterate over Features of type `Polygon`, you can store the id in the properties of the Polygon Feature.
From the code above, I see also that each `Polygon` component seems to create a new Map. I think this is not something you want isn't it?

**Comment by FerretAngel** [comment]
Sorry, just saw your reply now. Looks like there was a translation software glitch.
Let me rephrase the issue: I want to assign an ID to each polygon. 
Why? So I can easily tell which polygon the user has selected.
 I checked the documentation, but couldn't find any way to assign an ID.
I hope I can directly assign an ID to the polygon like this.
![image](https://github.com/MelihAltintas/vue3-openlayers/assets/58507654/75b8862e-68a0-452c-83bd-451c321ba2f3)
Or it could be like this to assign an ID to the feature.
![image](https://github.com/MelihAltintas/vue3-openlayers/assets/58507654/d1664c79-d2ac-451a-a299-eccd0a37e252)


> Please provide a minimal reproduction example. Ideally, you can fork the [Stackblitz starter template](https://stackblitz.com/edit/vue3-openlayers-template) for this purpose. From your code, I cannot extract what the `polygons` are and if they have an `id`. If you want to somehow iterate over Features of type `Polygon`, you can store the id in the properties of the Polygon Feature. From the code above, I see also that each `Polygon` component seems to create a new Map. I think this is not something you want isn't it?

**Comment by d-koppenhagen** [comment]
Hey, there is no ID Attribute but you can pass [properties](https://vue3openlayers.netlify.app/componentsguide/feature/) the the feature which could be `{ id: "my-id-1" }`. This properties are assigned to the underlying OpenLayers [Feature](https://openlayers.org/en/latest/apidoc/module-ol_Feature-Feature.html).
You can retrieve the properties later to distinguish from a Feature/FeatureLike like this: `feature.getProperties()?.id`.

**Comment by FerretAngel** [comment]
Great. That's what I need. 
Thank you very much for your patient answer!
